processing/src/main/java/org/apache/druid/query/JoinDataSource.java

```diff
@@ -476,10 +476,18 @@ private Function<SegmentReference, SegmentReference> createSegmentMapFunctionInt
                            .orElse(null)
                 )
             );
-
+            final Function<SegmentReference, SegmentReference> baseMapFn;
+            if (left instanceof JoinDataSource) {
```

Comment:
this seems worth a comment on what is going on. Is it still ok to do if left is not concrete?

Reply:
I'll add a comment as into why we are not using the isConcrete() check and instead using the instanceof check here

Reply:
Comment is added